packages/cli/src/util/createServer.ts

```diff
@@ -97,10 +108,25 @@ function pipeOutputToSignale(stream: Readable) {
   });
 }
 
-export type CreatePrismOptions = {
+function isProxyServerOptions(options: CreateBaseServerOptions): options is CreateProxyServerOptions {
+  return !options.dynamic && 'upstream' in options;
+}
+
+type CreateBaseServerOptions = {
   dynamic: boolean;
   cors: boolean;
-  host?: string;
+  host: string;
   port: number;
-  operations: IHttpOperation[];
+  document: string;
+  multiprocess: boolean;
+  reportViolations: 'stdout' | 'httpBody' | 'httpHeader';
```

Comment:
I increasingly thing we just need a single flag not a series of options. 

Would it be technically possible to have it always log to stdout (useful for prod), always report to headers (useful for everyone but not the most verbose format, so they can check the logs, and make an option that allows the body to show up?

Again we don’t want any camelCase in our values we don’t canelCase anywhere in our CLI (if we do it’s a bug we missed).

—fail or —errors or —panic or something, anything, I think might be easier.

Reply:
Also "report violations" is terminology I would use for sending actual failures to the stoplight API, which we might want to do independently of HTTP validation of "logging" (stdout).

Reply:
> If stdout is default then we don't need to offer it as a mode which can be turned on right? Should it not always be on?

Being default does not mean it's always on. If you do `httpHeader` (or `http-header`, whatever we'll come up with) — then stdout won't happen.

---

> Would it be technically possible and reasonable to 1) always log to stdout (useful for prod), 2) always report to headers (useful for everyone but not the most verbose format, so they can check the logs, and 3) make an option that allows the body to show up?

1) yes
2) yes
3) yes

Do you want me to do that? If so, what would be the option name? `--errors-for-violation`? Something else?

Reply:
> Being default does not mean it's always on. If you do httpHeader (or http-header, whatever we'll come up with) — then stdout won't happen.

yeah i was suggesting we should make it be default and always on.

<hr>

Yes. --errors is fine, it will be explained in the manual and we can get user feedback on what the name should be if people are unsure. `--errors-for-violation` is long and also assumes people know what violations are, which they dont have a reason to yet.

Reply:
> yeah i was suggesting we should make it be default and always on.

stdout? stdout and http headers?

Reply:
Logging is always happening in proxy just like in mocking, and warning is always happening too.

No, switches should be enabling a thing, so adding `--errors` would be turning it on.

Reply:
So to recap:

1. Always log stuff on the stdout ✅ 
2. Always send the `warning` header in the response ✅ 
3. `--errors` flag will decide whether the response should be returned back or "transformed into a 422 with the error detail"